### ISSUE TITLE
Spheres do not move to true edge when being moved by changing Mass

### ISSUE THREAD

**Issue by terracoda** [phetsims/gravity-force-lab#130]
@mbarlow12 while exploring phrasing for how describe the movement of the spheres when you are changing the mass, I noticed that neither sphere can be pushed to the edge.

- m1 only goes to 0.1 meter mark when m2's mass is changing and pushing m1 further left, and
- m2 only goes to 9.5 meter mark when m1's mass is changing and pushing m2 further right

Specifically, when changing mass of m1, m2 will stop at 9.5 meter mark and then further increases to m1's mass results in m1 moving backwards (left). The same is true when changing m2's mass when m1 is close to the left edge.

Is this is a visual bug?

And, I have a related question, how much should a sphere move when it is pushed? It seems to move 0.1 meter mark, but also seems to create more space in between, so that consecutive increases do not keep moving the sphere. Is this the intended behaviour? Please see the[ Edge-case Examples table](https://docs.google.com/document/d/1-37qAgde2XrlXBQae2SgjartM35_EnzDD9pdtd3nXAM/edit?ts=5c12c712#heading=h.kj3w5ar1mnks) in the design doc 

I am using Safari 12.0.2 and Mac OS 10.13.6 in case this is system related.

**Comment by mbarlow12** [comment]
The former is a result of how we're constraining the sphere positions to allow either sphere to move when radii are changing; I've added a small check that should allow the extra 0.1 on either end.

For the movement when pushed, we always constrain positions to the nearest 0.1 point (regardless of the specific unit). Since we're rounding the position values, there is some space (0 - 0.049999...) where the 'pushed' sphere won't move. Once the radius of one sphere is large enough to nudge the position of the other sphere 0.05 or more, the other sphere's position snaps to the next increment.

**Comment by terracoda** [comment]
@mbarlow12, thanks for the explanation, but I am still a little confused about what should be the proper behaviour, especially in the non-visual experience where the user is getting the position value (meter mark) on focus.

In the equation for calculating the force of gravity, it is the value for distance that matters, right? If I am understanding things properly, it is the distance that should never be equal to zero, but it should be fine if for position to be at the 0.0 meter mark. 

Since the position values are irrelevant to the equation, and the position range on the slider is 0.0 meter mark to 9.6 meter mark, a user should always be able to move or push a sphere to each of those extreme positions during interaction, right? 

In the non-visual experience, the extremes are not being reached in the case where a a sphere's changing mass is pushing the other sphere. That seems odd to me.

In terms of the rounding to 0.1 seems correct. It appeared to me while interacting that the spheres were jumping apart more than 0.1 meters. I'll test more on this once the default step size is set to 0.5 meters (https://github.com/phetsims/gravity-force-lab/issues/121).

I'll mark this issue for discussion at our meeting.

**Comment by terracoda** [comment]
Thanks @mbarlow12, nice work!
I verified this issue that the spheres can now be pushed to the edges.
Closing the issue.
